pkg/apis/operator/v1alpha1/types.go

```diff
@@ -188,6 +191,23 @@ type ControlPlane struct {
 // HighAvailability specifies the configuration settings for high availability for a resource.
 type HighAvailability struct{}
 
+// Kubernetes contains the version and configuration options for the Kubernetes components of the virtual garden
+// cluster.
+type Kubernetes struct {
+	// KubeAPIServer contains configuration settings for the kube-apiserver.
+	// +optional
+	KubeAPIServer *KubeAPIServerConfig `json:"kubeAPIServer,omitempty"`
+	// Version is the semantic Kubernetes version to use for the virtual garden cluster.
+	Version string `json:"version"`
+}
+
+// KubeAPIServerConfig contains configuration settings for the kube-apiserver.
+type KubeAPIServerConfig struct {
+	// KubeAPIServerConfig contains all configuration values not specific to the virtual garden cluster.
+	// +optional
+	*gardencorev1beta1.KubeAPIServerConfig `json:",inline"`
```

Comment:
Is this comma doing some magic or can it be removed?

```suggestion
	*gardencorev1beta1.KubeAPIServerConfig `json:"inline"`
```

Reply:
I don't think it can be removed. At least, this is the way we inline fields elsewhere: https://github.com/search?q=repo%3Agardener%2Fgardener%20%22%2Cinline%22&type=code

Reply:
I was confused about the leading `,`. Checking the topic in more detail, it seems like this is not even supported (https://github.com/golang/go/issues/6213), but widely used in the kubernetes space. Let's keep it as is.